breaking: 14.0.0 Release

jennifer-shehane opened the pull request:
BREAKING CHANGE: 14.0.0 release

# ⚠️ DO NOT SQUASH AND MERGE. USE A MERGE COMMIT

- https://github.com/cypress-io/cypress/issues/29925

## Checklist

### Before merging `release/14.0.0` into `develop`
- [x] Make sure all npm packages with breaking changes run against npm-release in circle CI. Do a dry run once ready to make sure major versions are applied to:
    - [x] @cypress/vite-dev-server
    - [x] @cypress/webpack-dev-server
    - [x] @cypress/react 
    - [x] @cypress/svelte
    - [x] @cypress/angular 
    - [x] @cypress/schematic
    - [x] Test the Cloud project on the binary https://github.com/cypress-io/cypress-services/issues/9730
- [x] The following packages are removed and need to be deprecated in npm
    - [x] @cypress/vue2
    - [x] @cypress/react18
    - [x] @cypress/angular-signals

### After merging `release/14.0.0` into `develop`
- [x] Make sure all npm packages with breaking changes are published as breaking changes:
    - [x] @cypress/vite-dev-server
    - [x] @cypress/webpack-dev-server
    - [x] @cypress/react 
    - [x] @cypress/svelte
    - [x] @cypress/angular 
    - [x] @cypress/schematic
- [x] Once major version is released, continue to give @cypress/schematic a full test with the angular cli
- [x] We need to make sure synk jobs now run on node 20 and not node 18

### Release day
- [x] Deprecate the [@cypress/angular-signals](https://www.npmjs.com/package/@cypress/angular-signals) package.  If we can add a message we just need to say the package merged upstream in [@cypress/angular](https://www.npmjs.com/package/@cypress/angular)
- [x] Deprecate the [@cypress/react18](https://www.npmjs.com/package/@cypress/react18) package. If we can add a message we just need to say the package merged upstream in [@cypress/react](https://www.npmjs.com/package/@cypress/react)
- [x] After binary is deployed to dev tag (step 10), create a PR into  https://github.com/cypress-io/cypress-component-testing-apps/pull/30  to bump all examples to use Cypress 14. While doing this:
    - [x]  Remove references to cypress/angular-signals and update them to cypress/angular.  smoke test the projects affected
    - [x] Remove references to cypress/react18 and update them to cypress/react. smoke test the projects affected
- [x]  After binary is deployed to dev tag (step 10), create a PR in https://github.com/cypress-io/component-testing-quickstart-apps/pull/18 to bump all  projects (4 in total) to use Cypress 14. Smoke test each project to make sure the Stepper components work
- [x] Merge Docs PR: https://github.com/cypress-io/cypress-documentation/pull/5964
- [x] Publish Cypress Blog (ping @jennifer-shehane to do this)
- [x] Ping Social Media to begin promotion (ping @jennifer-shehane to do this)
- [x] Ping internal newsletter team to tee up v14 email (ping @jennifer-shehane to do this)
- [x] Merge Component Testing Apps PR: https://github.com/cypress-io/cypress-component-testing-apps/pull/30 (needs to go in same time as Docs PR with release guide step 15.)
- [x] Merge Quickstart PR: https://github.com/cypress-io/component-testing-quickstart-apps/pull/18

## Details

- https://github.com/cypress-io/cypress/pull/30394
- https://github.com/cypress-io/cypress/pull/30307
- https://github.com/cypress-io/cypress/pull/30331
- https://github.com/cypress-io/cypress/pull/30390
- https://github.com/cypress-io/cypress/pull/30345
- https://github.com/cypress-io/cypress/pull/30456
- https://github.com/cypress-io/cypress/pull/30460
- https://github.com/cypress-io/cypress/pull/30463
- https://github.com/cypress-io/cypress/pull/30462
- https://github.com/cypress-io/cypress/pull/30641
- https://github.com/cypress-io/cypress/pull/29689
- https://github.com/cypress-io/cypress/pull/29688
- https://github.com/cypress-io/cypress/pull/29778
- https://github.com/cypress-io/cypress/pull/29680
- https://github.com/cypress-io/cypress/pull/30654
- https://github.com/cypress-io/cypress/pull/30651
- https://github.com/cypress-io/cypress/pull/30590
- https://github.com/cypress-io/cypress/pull/30633
- https://github.com/cypress-io/cypress/pull/30605
- https://github.com/cypress-io/cypress/pull/30585
- https://github.com/cypress-io/cypress/pull/30572
- https://github.com/cypress-io/cypress/pull/30558
- https://github.com/cypress-io/cypress/pull/30540
- https://github.com/cypress-io/cypress/pull/30512
- https://github.com/cypress-io/cypress/pull/30499
- https://github.com/cypress-io/cypress/pull/30489
- https://github.com/cypress-io/cypress/pull/30716
- https://github.com/cypress-io/cypress/pull/30700
- https://github.com/cypress-io/cypress/pull/30723
- https://github.com/cypress-io/cypress/pull/30696
- https://github.com/cypress-io/cypress/pull/30770

jennifer-shehane committed:
init changelog

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30306)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30321)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30322)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
breaking: Remove support for Node.js 16, Node.js 21, and glibc < 2.28 (#30307)

* Update minimum node engines

BREAKING CHANGE: Remove Node.js 16 and Node.js 21 support

* update changelog

* change entry to addresses

* trigger publish-binary pipeline on appropriate branch

* remove better-sqlite code

* Update changelog

* changelog

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
Merge branch 'release/14.0.0' of https://github.com/cypress-io/cypress into release/14.0.0

jennifer-shehane committed:
Run on release/14.0.0 branch

jennifer-shehane committed:
cache bump

jennifer-shehane committed:
remove instruction to update electron version in centos 7 step

ryanthemanuel committed:
chore: fix better-sqlite3 rebuild process (#30338)

* chore: fix better-sqlite3 rebuild

* Update cache-version.txt

jennifer-shehane committed:
Merge branch 'release/14.0.0' of https://github.com/cypress-io/cypress into release/14.0.0

jennifer-shehane committed:
Revert "remove instruction to update electron version in centos 7 step"

This reverts commit 01131cfecaab41acdc6b678fddf39515c2301889.

jennifer-shehane committed:
Remove electron version instructions for adding to step that no longer exists

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30346)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30355)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
breaking: remove log:memory:pressure and firefox:force:gc APIs (#30331)

* remove foxdriver and force:gc/log:memory:pressure APIs

* Add changelog entry

* remove patched foxdriver file

* define types/spec FoundSpec without ref to Cypress namespace

* args

* chore: update unit test to still throw firefox cannot connect error if CDP fails to connect instead of the rtrying socket, which is no longer used when connecting to firefox

* Remove firefox-memory system test

---------

Co-authored-by: Cacie Prins <cacie@cypress.io>
Co-authored-by: AtofStryker <bglesias@gmail.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30356)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30357)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

mschile committed:
chore: build better-sqlite3 with glibc 2.28 (#30390)

ryanthemanuel committed:
chore: merge develop

ryanthemanuel committed:
blank -- run ci

ryanthemanuel committed:
fix merge

ryanthemanuel committed:
fix merge

ryanthemanuel committed:
Update .circleci/cache-version.txt

ryanthemanuel committed:
Merge pull request #30423 from cypress-io/ryanm/chore/merge-develop

chore: merge develop to release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30424)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

ryanthemanuel committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

ryanthemanuel committed:
Merge branch 'develop' into release/14.0.0

ryanthemanuel committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30439)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>
Co-authored-by: Ryan Manuel <ryanm@cypress.io>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
breaking: Remove `run-ct`and `open-ct` CLI commands (#30456)

* Remove run-ct and open-ct as CLI commands

BREAKING: Remove run-ct and open-ct as CLI commands

* Add changelog entry

* add CLI to changelog

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
breaking: remove deprecated before:browser:launch signature code that allowed Arrays (#30460)

* chore: release 13.15.1 (#30454)

* chore: Update Chrome (beta) to 131.0.6778.13 (#30451)

Co-authored-by: cypress-bot[bot] <41898282+cypress-bot[bot]@users.noreply.github.com>

* remove deprecated before:browser:launch signature code

* Update changelog

* Update snapshots & failing tests

BREAKING: remove deprecated before:browser:launch signature code that allowed Arrays

* Add back options

* update test spec to use non deprecated before:browser:launch signature

* removed test for deprecated before:browser:launch signature

---------

Co-authored-by: Bill Glesias <bglesias@gmail.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: cypress-bot[bot] <41898282+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
breaking: Remove experimentalFetchPolyfill flag (#30466)

* remove experimentalFetchPolyfill

BREAKING: remove experimentalFetchPolyfill

* Add changelog entry

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30480)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30485)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

ryanthemanuel committed:
chore: upgrade electron to 32.2.0 (#30394)

* chore: upgrade electron to 32.2.0 -- run ci

* fix node versions

* fix build

* fix evil-dns

* various fixes

* fix srcdoc

* various fixes

* update package.json

* update yarn.lock

* various fixes

* fix integrity check

* fix binary verification

* various fixes

* update yarn.lock

* fix typo

* fix lock file

* fix tests

* fix tests

* various fixes

* various fixes

* various fixes

* fix stuff

* fix things

* try to fix errors

* fix

* updates

* add yarn berry

* upgrade electron

* attempt with gcc

* blank

* update arm64 executor

* try to fix the arm64 issue

* fix arm64

* bump cache

* attempt to fix arm64 again

* attempt to fix arm64 again

* fix darwin problems

* merge release/14.0.0

* update yarn.lock

* blank -- run ci

* blank

* blank

* blank

* try something

* try something

* clean up

* blank

* try to fix sessions issue

* fixes

* get more info

* get more info

* try something

* fix

* try something

* try something

* try something

* tweak

* one more thing

* let us see if this works

* blank

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* Apply suggestions from code review

* Apply suggestions from code review

* Update cli/CHANGELOG.md

Co-authored-by: Jennifer Shehane <jennifer@cypress.io>

* Update cli/CHANGELOG.md

Co-authored-by: Jennifer Shehane <jennifer@cypress.io>

* Update CHANGELOG.md

* Update cache-version.txt

* fix nx (maybe)

* Update package.json

* try something

* Update packages/app/src/runner/aut-iframe.ts

* bump version

* try to bust cache

* try to invalidate cache

* Update cli/CHANGELOG.md

* suppress benign warnings

* Apply suggestions from code review

* PR comments

* PR comment

* Apply suggestions from code review

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* Update cli/CHANGELOG.md

Co-authored-by: Matt Schile <mschile@cypress.io>

* update debug scenario 4 regex

* Update scripts/binary/trigger-publish-binary-pipeline.js

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>
Co-authored-by: Jennifer Shehane <jennifer@cypress.io>
Co-authored-by: Matt Schile <mschile@cypress.io>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30490)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
breaking: Remove deprecated delayMs option of cy.intercept (#30463)

* Remove deprecated delayMs property from cy.intercept

BREAKING: Remove deprecated delayMs property from cy.intercept

* Update test to new signature

* Add changelog entry

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30491)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

AtofStryker committed:
breaking: dropping support for webpack-dev-server 3 from @cypress/webpack-dev-server and make `@cypress/webpack-dev-server` ship with `webpack-dev-server` version 5 by default (from 4) (#30467)

* breaking: dropping support for webpack-dev-server 3 from @cypress/webpack-dev-server

    BREAKING CHANGE: support for webpack-dev-server version 3 has been removed. webpack-dev-server 3 is no longer maintained by the webpack team. To reduce overhead, Cypress is now removing support in Cypress 14.

* update create-react-app-configured/unconfigured

* updated vueclivue2-configured to use vue cli v5

* updated vueclivue2-unconfigured to use vue cli v5

* updated vueclivue3-unconfigured to use vue cli v5

* updated vueclivue3-configured to use vue cli v5

* updated vueclivue3-custom-index-html to use vue cli v5

* update react17 project to use webpack-dev-server v4 (default ships v5 which does not work with webpack 4)

* update react18 project to use webpack-dev-server v4 (default ships v5 which does not work with webpack 4)

* update nuxt version to avoid some issues in the preprocessed webpack config

* update cy in cy component scaffolding tests to run on more suites, as well as skip config warning for outdated vuecli3 as it will throw due to wds being on version 3. this is a VERY old version, and the test needs to be updated to use vuecli5 so we get wds v5 (vuecli5 support will be deprecated)

* no longer support cra v4 and vue/cli-service v4

* remove cra v4 and vue cli service v4 detect tests as they are no longer supported [run ci]

* chore: update cra app custom index to use cra 5 [run ci]

* update protocol spec snapshot [run ci]

* chore: remove stale references to old vuecli system tests [run ci]

* update webpack-dev-server-4 to actually be wds 4 [run ci]

* add changelog entry [run ci]

* update cache key and tests that should not pass [run ci]

* add issue to changelog entry

* no longer clean up default-gateway as it was shipped with webpack-dev-server 4, which is now a dev dependency [run ci]

* fix issues with node-ipc being out of date [run ci]

* move changelog to correct section [run ci]

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30500)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
breaking: remove logic around min supported browsers (#30462)

* remove logic around min supported browsers

BREAKING: remove logic around min supported browsers

* changelog entry

* forgot to save some changed files

* update broken tests

* remove another test testing unsupported browsers

* Add back logic for browser validator

* return if validator is undefined

* update called browser

* fix failing tests

* fix logic of unsupported showing

* add parens

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30513)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
breaking: remove support for vite 2 and 3 from `@cypress/vite-dev-server` (#30489)

* chore: update system tests for vite 4 and 5 to latest [run ci]

* chore: remove projects vite 2.8.6, 2.9.1, vite3.0.2 as they are no longer supported by @cypress/vite-dev-server

* chore: update vite-ct-function-api system tests to latests

* chore: update vite-ct-object-api system tests to latests

* chore: update vite-simple system tests to latests

* breaking: remove support for vite 2 and 3 [run ci]

* update react-vite-ts-configured to latest deps

* update react-vite-ts-unconfigured to latest deps

* update vue2 to latest (to be removed later in v14)

* update vue3-vite-ts-configured to latest deps (need to fix typings issue within vue)

* update vue3-vite-ts-unconfigured to latest deps

* update vue3-vite-ts-custom-index-html to latest deps (need to fix typings issue within vue)

* update vue3 to latest deps

* update detection spec to be compatible with latest compatible deps. removed test for allowing passing scaffolding for installing prereleases as we don't support prereleases ahead of time

* add changelog

* chore: update react-vite-ts-configured

* update svelte vite

* update ct-public-api-solid-js

* update system test vite dependencies

* update comments on typings issues and missed dependencies [run ci]

* add addresses section [run ci]

* update lock file for component-testing-outdated-dependencies [run ci]

* update lock file [run ci]

* update missed yarn lock file [run ci]

* fix issues inside vue2 project package not using correct dependencies [run ci]

* fix issues with failing vds test [run ci]

* fix failing tests in vite-dev-server by fixing source map references (expected since we are on 3 major versions ahead of vite) and install @cypress/vite-dev-server in correct packages [run ci]

AtofStryker committed:
breaking: remove nuxt2 support from Cypress Component Testing [run ci] (#30499)

jennifer-shehane committed:
dependency: Update jQuery to latest (#30345)

* remove jquery patch

BREAKING: remove the patch that was preventing width/height calculation changes

* changelog entry

* fix issue link

* Update halfScrollPixels to width/height that jQuery returns

* Update scroll tests to properly check for half with no scrollbar

* remove passing variable that doesn't exist in jquery scrollto

* bump to jQuery 3.5.0 + fix invalid DOM in tests

* bump down jquery - got too excited

* Fix the tests to be accurate

* bump jquery

* switch to client, see what breaks

* Repatch jquery and update tests back to old logic

* add dep changelog entry

* remove test that's reliant on deprecated jQuery focus

* update test to follow cypress conventions for chaining

* Update regex to include jQuery stack traces as well as sizzle since jQuery is moving away from sizzle

* update changelog

* update focus link to fix it

* added patch for :focus selector

* Move changelog entry to 4.0

---------

Co-authored-by: Matthew Schile <mschile@cypress.io>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30535)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30537)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

AtofStryker committed:
breaking: Cypress Component Testing no longer supports vue2 [run ci] (#30502)

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30543)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
deprecation: Deprecate resource type of intercept's request object (#30512)

* deprecation: deprecate resourceType of intercept's request object

* Add changelog entry

AtofStryker committed:
breaking: remove support for Next.js 10, 11, 12, and 13 for Cypress component Testing [run ci] (#30506)

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30544)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

senpl committed:
fix: #29605 for display: contents parent visibility with correct width check (#29680)

* forst working version

* remove visibilityVisible as non essencial

* changelog for pipeline fix

* fix only for firefox. It works in other browsers without this change

* changelog update for pipeline

* pipeline fix

* return of previous spacing

* Update CHANGELOG.md

* changelog without unnecessery changes

* changelog fix

* fix changelog entry

* update test to properly have a test for the issue

* add test for other elements to not be visible

* bump cache

---------

Co-authored-by: Jennifer Shehane <jennifer@cypress.io>

AtofStryker committed:
Merge branch 'develop' into chore/merge_develop

AtofStryker committed:
breaking: remove support for angular 13,14,15, and 16. minimum version is now 17.2.0 (#30539)

breaking: cypress-schematic only supports angular 17.2.0 and up [run ci]

fix updates missed [run ci]

AtofStryker committed:
Merge branch 'release/14.0.0' of github.com:cypress-io/cypress into chore/merge_develop

ryanthemanuel committed:
Merge pull request #30559 from cypress-io/chore/merge_develop

chore: merge `develop` into `release/14.0.0`

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30564)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30565)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30566)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache (#30562)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>
Co-authored-by: Jennifer Shehane <jennifer@cypress.io>

AtofStryker committed:
breaking: remove the cypress/angular-signals test harness. The changes incorporated in @cypress/angular-signals are now merged upstream into @cypress/angular and the package is no longer needed [run ci] (#30540)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30576)

* chore: updating v8 snapshot cache

* chore: updating v8 snapshot cache

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>
Co-authored-by: Ryan Manuel <ryanm@cypress.io>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30577)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30578)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30579)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30581)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30582)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30583)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
breaking: remove support for create react app for component testing as it is no longer maintained [run ci] (#30558)

mschile committed:
breaking: remove Svelte 3 support (#30572)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30595)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30596)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30597)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
breaking: remove support for @vue/cli-serivce from Cypress Component Testing [run ci] (#30585)

mschile committed:
breaking: remove typescript v3 from the configuration wizard (#30605)

AtofStryker committed:
chore: update to react 18 for the reporter from 17.0.2 to 18.3.1 [run ci] (#30567)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30612)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30613)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30615)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
chore: update system tests to use react 18 and install latest dependencies for applicable system tests. use react18 harness before removal [run ci] (#30614)

AtofStryker committed:
breaking: remove support for React 16 and 17 for Cypress Component Testing. Additionally, remove the `cypress/react18` testing harness and merge it upstream with `cypress/react` (#30590)

* breaking: remove support for react 16 and 17 for component testing and move cypress/react18 upstream into cypress/react [run ci]

* update tests / suggestions from code review [run ci]

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30625)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30627)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30626)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

MikeMcC399 committed:
dependency: update @electron/rebuild to 3.7.1 (#30633)

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30638)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30637)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30639)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
feat: support react 19 for component testing with cypress/react [run ci] (#30651)

ryanthemanuel committed:
chore: suppress EGL Driver message in electron 32 (#30663)

* chore: suppress EGL Driver message in electron 32

* remove bad files

AtofStryker committed:
breaking: make JustInTimeCompile GA from experimentalJustInTimeCompile and default to true (#30641)

* pull support for JIT from vite since it doesn't have an affect for end users or performance (webpack only) [run ci]

rebase this

* remove the experimentalJIT flag and make it GA (default still false) [run ci]

* enable justInTimeCompile by default [run ci]

* chore: bump cache [run ci]

AtofStryker committed:
feat: support next 15 [run ci] (#30654)

MikeMcC399 commented:
- When the `release/14.0.0` branch is rebased on `develop` I'll submit a documentation update to [CONTRIBUTING > Requirements](https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#requirements) as a follow-on to PR #30633

jennifer-shehane committed:
merge develop

senpl committed:
fix: update visibility to support parent overflow:clip without height/width (#29778)

senpl committed:
fix: update visibility for elements with textContent but without width/height (#29688)

senpl committed:
fix: visibility issue with parent absolute (#29689)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30689)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30690)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30691)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

MikeMcC399 committed:
docs: remove node-gyp contributor workarounds / restrictions (#30671)

mschile committed:
test: remove duplicate add function (#30693)

AtofStryker committed:
feat: support angular 19 and update tests [run ci] (#30675)

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

mschile committed:
misc: remove component testing API stubs that were removed in Cypress 11 (#30696)

AtofStryker committed:
feat: component testing now supports vite 6 [run ci] (#30679)

mschile committed:
test: update CI to run `npm/vue` tests (#30707)

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30710)

* chore: updating v8 snapshot cache

* index on release/14.0.0: 09c9b2ff5b test: update CI to run `npm/vue` tests (#30707)

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30709)

* chore: updating v8 snapshot cache

* index on release/14.0.0: 09c9b2ff5b test: update CI to run `npm/vue` tests (#30707)

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30708)

* chore: updating v8 snapshot cache

* index on release/14.0.0: 09c9b2ff5b test: update CI to run `npm/vue` tests (#30707)

---------

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
Merge branch 'develop' into release/14.0.0

mschile committed:
dependency: electron 33 upgrade (#30716)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30724)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30725)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30726)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
breaking: remove support for svelte 4 and support svelte 5 [run ci] (#30700)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30734)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30735)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30736)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
chore: remove support for react 19 RC and make Nextjs 15.0.4 the minimum next 15 version supported [run ci] (#30723)

Co-authored-by: Jennifer Shehane <jennifer@cypress.io>

jennifer-shehane committed:
merge develop

jennifer-shehane committed:
merge develop

jennifer-shehane committed:
yarn.lock update with 1.22.22

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30746)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30747)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30748)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

mschile committed:
test: fix default size system test (#30759)

AtofStryker committed:
chore: bump @vue/test-utils from 2.3.2 to 2.4.6 [run ci] (#30739)

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30780)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30781)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30782)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Merge branch 'develop' into release/14.0.0

jennifer-shehane committed:
Rearrange changelog a bit

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30799)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30800)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30801)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

jennifer-shehane committed:
Add missing parens in example of fetch update

jennifer-shehane committed:
remove minSupportedVersion from browser (snuck in from dev merge)

jennifer-shehane committed:
fix merge from dev snapshot

ryanthemanuel committed:
blank

ryanthemanuel committed:
chore: fix various system test build failures (#30814)

* chore: fix various system test build failures

* rename tests

cacieprins committed:
breaking: no longer inject document.domain by default (#30770)

* remove experimentalSkipDomainInjection, add and deprecate injectDocumentDomain

* remove experimentalSkipDomainInjection, add injectDocumentDomain

* begin rethreading domain injection

* complete document domain transition

* move some cookie specs to separate test run

* origin and privileged commands with default docdom inject

* fix privileged channel when injecting document domain

* rm unnecessary .getOrigin abstraction in cors lib

* move remote-states in prep for refactor Replace Conditional with Polymorphism

* refactor remote states to strategy pattern

* cookie commands work as expected w cross origin bridge on different origins

* some origin tests updated

* run tests with document domain enabled

* run tests actually

* use correct config, swap conditional

* check-ts

* inject documetn domain for webkit tests

* do not exec injectDocumetnDomain in parallel

* fix ServerBase construction in tests to include cfg now

* pass cfg to ServerBase

* improved integration tests

* remove document domain checks for all server integration specs - will add injectDocumentDomain cases

* tests for injecting document domain when configured to

* square away server integration tests

* ensure cookies are set correctly, potentially

* errors pkg snapshots

* fix config tests

* fixing config tests

* somewhat improves tests for cors policies in packages/network

* fix ts err in server-base

* enable injectDocumentDomain for cy in cy tests

* fix Policy type ref

* refactor cypress-schematic ct spec to be less prone to timeouts

* run vite-dev-server tests with injectDocumentDomain

* rm document domain assertion from page_loading system test

* add system tests that test with injectDocumentDomain and others that test with cy.origin

* fix results_spec snapshot

* update experimentalSkipDomainInjection system test

* different behavior for certain net_stubbing tests based on injectDocumentDomain or not

* fix ts

* extract origin key logic from remote states, for now

* move server-base and response-middleware over to new pattern

* WIP - reentry

* fix build, remove console.log

* check-ts

* fix spec frame injection

* remove injection for localhost

* mostly fix vite-dev-server app integration tests

* fix codeframe in certain cases in chrome

* drop internal stack frames from stacks intended for determining code frame data

* some improvements to vite ct error codeframes

* fix proxy unit tests to use document domain injection util class

* rm .only

* fix all vite ct error specs

* rm console.log

* slight refactor to util class to make easier to test

* fix refactor - missing rename in files.js

* several tests do not set testingtype in config, so just check against component instead of checking for e2e

* revert changes to getInvocationDetails to see if that breaks tests

* re-enable stack stripping in invocation details for chrome

* new snapshots with more accurate invocation details

* test for same-site cross-origin cookie behavior

* ignore window.top ts errors

* revert forcing injectDocumentDomain in vite-dev-server cy config

* fix normalized whitespace for firefox "loading_failed" error

* always trim trailing wsp from stack before appending additional content

* force normalization of whitespace to three \n when adding additional stack details

* normalize wsp between stack and additional stack to "\n  \n" in firefox

* remove stack_utils attempt at normalizing wsp

* various cleanup: remove commented console logs, add more detailed comments

* add on links to error messages

* remove experimentalSkipDomainInjection from exported type defs

* Update system-tests/test/experimental_skip_domain_injection_spec.ts

Co-authored-by: Bill Glesias <bglesias@gmail.com>

* Update packages/driver/cypress/e2e/e2e/origin/cookie_misc.cy.ts

Co-authored-by: Bill Glesias <bglesias@gmail.com>

* no need to coerce a boolean value to a booleanc

* export base config from primary cypress config in driver for use in inject-document-domain test subset

* lift experimentalSkipDomainInjection breaking option to root

* rollback config/options changes

* rm invalid comment

* use hostname instead of origin to create cookie from automation cookie

* clarify stack regex in results_spec

* lint

* take a stab at the changelog entries for this

* Update cli/CHANGELOG.md

Co-authored-by: Ryan Manuel <ryanm@cypress.io>

* Update cli/CHANGELOG.md

Co-authored-by: Ryan Manuel <ryanm@cypress.io>

* reenable locally-failing test

* changelog

* snapshot updatesfor experimental skip domain injection err msg

* remove packageManager declaration in package.json

---------

Co-authored-by: Bill Glesias <bglesias@gmail.com>
Co-authored-by: Jennifer Shehane <jennifer@cypress.io>
Co-authored-by: Ryan Manuel <ryanm@cypress.io>

github-actions[bot] committed:
chore: Update v8 snapshot cache - darwin (#30823)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - linux (#30824)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: Update v8 snapshot cache - windows (#30825)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

cacieprins committed:
refactor: clean up experimentalSkipDomainInjection removal (#30826)

* chore: clean up experimentalSkipDomainInjection removal

* rm corepack yarn line

* update snapshots

* rm dead code, update error message

jennifer-shehane committed:
merge develop

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30828)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30829)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore: updating v8 snapshot cache (#30830)

Co-authored-by: cypress-bot[bot] <+cypress-bot[bot]@users.noreply.github.com>

AtofStryker committed:
breaking: release version 3.0.0 of @cypress/angular.

BREAKING CHANGE: Cypress 14 drops support for Angular 13-16. Angular 17.2 and up is supported.

AtofStryker committed:
breaking: release version 3.0.0 of @cypress/schematic.

BREAKING CHANGE: Cypress 14 drops support for Angular 13-16. Angular 17.2 and up is supported.

AtofStryker committed:
breaking: release version 9.0.0 of @cypress/react.

BREAKING CHANGE: Cypress 14 drops support for React 16-17. React 18 and up is supported.

AtofStryker reviewed (changes requested):
semantic commits for npm release are not for npm packages. in the process of fixing this but until fix DO NOT MERGE this PR

AtofStryker committed:
breaking: release version 3.0.0 of @cypress/svelte.

BREAKING CHANGE: Cypress 14 drops support for Svelte 3-4. Svelte 5 is supported.

AtofStryker committed:
breaking: release version 6.0.0 of @cypress/vite-dev-server.

BREAKING CHANGE: Cypress 14 drops support forVite 2-3. Vite 4 and up is supported.

AtofStryker committed:
breaking: release version 4.0.0 of @cypress/webpack-dev-server.

BREAKING CHANGE: Cypress 14 drops support for webpack-dev-server 3. webpack/webpack-dev-server 4 and 5 are supported. @cypress/webpack-dev-server also ships with webpack-dev-server v5 by default.

AtofStryker reviewed (approved):
package versions are now correct and will be published correctly on merge commit into develop 
<img width="305" alt="Screenshot 2025-01-07 at 5 10 14 PM" src="https://github.com/user-attachments/assets/3699daa9-6604-4510-ab9d-e6784d4479d1" />
